Pearson-Advance/frontend-app-learning
# PADV-864 - feat: avoid self enrollments.

## Description
There are no sites that should allow self enrollments. Currently, you can go to `http://localhost:2000/course/{course-id}/home` and you will be able to enroll in the course:

![image](https://github.com/Pearson-Advance/frontend-app-learning/assets/40271196/709b069e-26cd-4bab-96b0-7425c8cc7f2d)

This behavior is not desired for our business logic, since anyone could create an account in any site and when trying `http://localhost:2000/course/{course-id}/home`, then this user could self enroll.

## Changes
- [ ] Add `DISABLE_SELF_ENROLLMENT` runtime configuration variables
- [ ] Use `DISABLE_SELF_ENROLLMENT` to extend `canEnroll` behavior.

## How to test?

1. Go to `http://localhost:2000/course/{course-id}/home` and check you can see the enroll buttons.
2. Checkout to  this branch.
3. Add `"DISABLE_SELF_ENROLLMENT": "true"` to `MFE_CONFIG` in the site configuration to the site you are testing
4. Go to  `http://localhost:2000/course/{course-id}/home`
5. You won't see the enroll buttons.
![image](https://github.com/Pearson-Advance/frontend-app-learning/assets/40271196/0ce7c04e-a84e-49f7-ad16-5952c209fa5c)

## Ticket:
[PADV-864](https://agile-jira.pearson.com/browse/PADV-864)
